src/lib/components/Modal/_hooks/useModalScrollPrevention.js

```diff
@@ -0,0 +1,34 @@
+import { useLayoutEffect } from 'react';
+
+export const useModalScrollPrevention = (preventScrollUnderneath) => {
+  useLayoutEffect(
+    () => {
+      if (preventScrollUnderneath === 'off') {
+        return () => {};
+      }
+
+      if (preventScrollUnderneath === 'default') {
+        const scrollbarWidth = Math.abs(window.innerWidth - window.document.documentElement.clientWidth);
+        const prevOverflow = window.document.body.style.overflow;
+        const prevPaddingRight = window.document.body.style.paddingRight;
+
+        window.document.body.style.overflow = 'hidden';
+        if (Number.isNaN(parseInt(prevPaddingRight, 10))) {
```

Comment:
Should be this condition documented in `[Prevent scrolling underneath the Modal](http://localhost:3000/components/modal#prevent-scrolling-underneath-the-modal)` section? What does @adamkudrna think?

Reply:
We could mention the scroll prevention accounts with scrollbar if there is one.

Reply:
Is there a use case when the user would prefer other behavior? If not then in my view there is no need to mention it.

It is expected that these things are handled correctly and there is no need to point it out. The shorter the documentation the greater chance that someone will read it.

Reply:
Well, for the other use cases there are options to turn it off or provide custom functions. Since I already mentioned each mode in the docs, I can add few words to point out that we account for scrollbar width. If someone wants to know how (and if it is the right behaviour for his use case), he can look into it in the code.

Reply:
I think it not necessary to add anything to the docs, but do as you want. I have no strong feelings about it.